infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -481,6 +469,11 @@ private synchronized void startSnapshotApplyAsync(LogReplication.LogReplicationE
 
     private synchronized void startSnapshotApply(LogReplication.LogReplicationEntryMsg entry) {
         log.debug("Entry Start Snapshot Sync Apply, id={}", entry.getMetadata().getSyncRequestId());
+        //set data_consistent as false and clear streams which has evidenced data from active and the streams
+        // which have been locally written (aimed for replication) but yet were not replicated from active to standby (empty on active)
+        // Note: the locally updated streams must be cleared or we could pollute the state of the DB
+        setDataConsistentWithRetry(false);
```

Comment:
I see this method just called once, why isn't it set back through the same method? or where is it set back?

Unless I missed something, then it means we don't have a test to validate that the flag is properly set/unset.

Reply:
We set it back to true from the sinkManager itself..via `completeSnapshotApply()`, and which in turn calls `logReplicationMetadataManager.setSnapshotAppliedComplete(entry);`

The reason we don't use the same method is because they set different metadata :
1. setDataConsistentWithRetry() sets dataConsistent to false and nothing more.
2. setSnapshotAppliedComplete() sets few other metadata along with dataConsistent to true